### DISCUSSION TITLE
Compiling from git repo build.sh

### DISCUSSION THREAD

**Discussion by tathastu871** [termux/termux-packages#8287] [Q&A]
trying to compile a program. It has no release.
in build.sh if i enter SRCURL to repourl.git
it keeps throwing warning: Could not find remote branch v to clone.
fatal: Remote branch v not found in upstream origin
and keeps asking sha256sum how to get sha256 for repo.???

Below is build.sh
```
TERMUX_PKG_HOMEPAGE=https://github.com/ericchiang/xpup
TERMUX_PKG_DESCRIPTION="command line tool for processing XML"
TERMUX_PKG_LICENSE="MIT"
TERMUX_PKG_MAINTAINER="@termux"
TERMUX_PKG_VERSION=
TERMUX_PKG_SRCURL=https://github.com/ericchiang/xpup.git
TERMUX_PKG_AUTO_UPDATE=true

termux_step_make() {
	termux_setup_golang

	cd "$TERMUX_PKG_SRCDIR"

	export GOPATH="${TERMUX_PKG_BUILDDIR}"
	mkdir -p "${GOPATH}/src/github.com/ericchiang/"
	cp -a "${TERMUX_PKG_SRCDIR}" "${GOPATH}/src/github.com/ericchiang/xpup"
	cd "${GOPATH}/src/github.com/ericchiang/xpup"
	export GO111MODULE=off

	go mod init
	go mod tidy
	go build
}

termux_step_make_install() {
	install -Dm700 -t "$TERMUX_PREFIX"/bin "$GOPATH"/src/github.com/ericchiang/xpup/xpup
}
```

**Comment by truboxl** [comment]
If your `TERMUX_PKG_SRCURL=` is a git repo, you need to add `TERMUX_PKG_GIT_BRANCH=`

The git branch can be `master`, `main` or something else, check that git repo of yours

You dont need to add `TERMUX_PKG_SHA256=` if `TERMUX_PKG_SRCURL=` is a git repo
